[junhonglow] iP

junhonglow opened the pull request:
### Bocchi 🎸 

> I don't want to work. - Hitori Gotoh ([source](https://youtu.be/dQw4w9WgXcQ))

Bocchi frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

To begin using it:

1. Download it from [here](https://github.com/junhonglow/ip)
2. Add tasks to it
3. Go wild

And it is **FREE**!

Features:

- [x] Managing Tasks
- [x] Managing Deadlines
- [ ] Reminders (Coming soon)

If you program as well, you can use it to practice Java too. Here's the ``main`` method:
```Java

public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

junhonglow committed:
Modify name of chatbot

junhonglow committed:
Add greeting and exit message

junhonglow committed:
Add private methods for greeting and exit

junhonglow committed:
Add echo functionality

junhonglow committed:
Add functionality to create task

junhonglow committed:
Add functionality to maintain a list of tasks

junhonglow committed:
Modified chatbot to implement new functions

junhonglow committed:
Add functionality to mark/unmark tasks

junhonglow committed:
Modified chatbot to implement mark/unmark task functionality

junhonglow committed:
Modify Task and mark/unmark methods to be abstract

junhonglow committed:
Add Todo, Deadline and Event classes

junhonglow committed:
Modify logic of string parsing and add functionality of new types of tasks

junhonglow committed:
Add missing lines after adding tasks

junhonglow committed:
Fix typo

junhonglow committed:
Add semi-automated testing

junhonglow committed:
Fix docstring

junhonglow committed:
Added exception handling

junhonglow committed:
Modify name of function to be more consistent

junhonglow committed:
Add functionality to delete tasks

junhonglow committed:
Reformat code

junhonglow committed:
Create task package

junhonglow committed:
Add loading/saving functionality

junhonglow committed:
Merge branch 'branch-Level-7'

junhonglow committed:
Add DateTime Functionality to tasks

junhonglow committed:
Update tests

junhonglow committed:
Merge branch 'branch-Level-8'

junhonglow committed:
Add Ui and Parser classes

junhonglow committed:
Merge branch 'branch-A-MoreOOP'

junhonglow committed:
Add packages

junhonglow committed:
Merge branch 'master' into add-gradle-support

junhonglow committed:
Configure gradle

junhonglow committed:
Add JUnit tests

junhonglow committed:
Add javadoc comments

junhonglow committed:
Fix code to adhere to coding standards

junhonglow committed:
Merge branch 'branch-A-JavaDoc'

junhonglow committed:
Merge branch 'branch-A-CodingStandard'

junhonglow committed:
Add find functionality

junhonglow committed:
Merge branch 'branch-Level-9'

keaganpzh commented on src/main/java/Bocchi.java:
```diff
@@ -0,0 +1,61 @@
+
+    /**
+     * Runs the program
+     */
```
Perhaps you can abstract this `run()` method further, e.g. make an `initialise()` method to instantiate your scanner and storage etc. That way `run()` will only contain logic relevant to the uptime of the program.

keaganpzh commented on src/main/java/exceptions/EmptyTaskException.java:
I like how you made your own exceptions extensively.

keaganpzh commented on src/main/java/parser/Parser.java:
Your parsing of user inputs is nicely done. Wow!

keaganpzh commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,166 @@
+    private void displayTasks(TaskList taskList) {
+        ui.displayTasks(taskList);
+    }
+
```
Missing header comment?

keaganpzh reviewed (commented):
Nicely done! Your code is clean and readable.

jason-raiin commented on src/main/java/exceptions/BocchiException.java:
Very nice use of exceptions with detailed and instructive error messages.

jason-raiin commented on src/main/java/exceptions/InvalidSyntaxException.java:
```diff
@@ -0,0 +1,22 @@
+        case "event":
+            return "Please use the following syntax when creating an event:\n"
+                    + "{activity} /from {start time} /to {end time}";
+        }
+        return "Unexpected error occurred in exceptions.InvalidSyntaxException!";
```
Missing default case

```suggestion
        default: 
            return "Unexpected error occurred in exceptions.InvalidSyntaxException!";
        }
```

jason-raiin commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,166 @@
+            task = new Event(eventTokens[0], eventTokens[1], eventTokens[2]);
+            break;
+        // Task.Todo is the default case
+        default:
+            task = new Todo(input);
```
Default case should contain a `break;` statement as well, according to the coding standard.

jason-raiin reviewed (commented):
Overall very nice OOP and extensive documentation of methods and detailed exception messages. Should be careful of the coding standards for `switch` statements.

junhonglow committed:
Add gradle dependencies for gui

junhonglow committed:
Add gui classes

junhonglow committed:
Add resources for gui

junhonglow committed:
Modify classes to use gui

junhonglow committed:
Merge branch 'branch-Level-10'

junhonglow committed:
Add checkstyle configs

junhonglow committed:
Fix application name

junhonglow committed:
Update gradle build
Change application to use Launcher class instead of Bocchi class
Allow assertions

junhonglow committed:
Add functionality to check task equality

A static task counter was included because two tasks could have the same name
but we do not want them to be considered equal.

junhonglow committed:
Add assertions to add/delete

This includes an additonal method in TaskList to check if a Task exists within it

junhonglow committed:
Remove magic literals in error classes
Error messages are used directly in the return values

Should the error messages be reused in multiple sections, it will
allow easier changes in the future.

Let's change all error messages into constants.

Note that %s is used in some messages to allow for string formatting.

junhonglow committed:
Apply SLAP principle to Bocchi class

Bocchi has logic implemented to process and load the save file.

There are multiple layers of abstraction within the same method which is not preferred.

Let's abstract the logic into the BocchiLoader class.

junhonglow committed:
Merge pull request #1 from junhonglow/branch-A-Assertions

Branch a assertions

junhonglow committed:
Merge branch 'master' into branch-A-CodeQuality

junhonglow committed:
Merge pull request #2 from junhonglow/branch-A-CodeQuality

Branch a code quality

junhonglow committed:
Add conflict checks for tasks

junhonglow committed:
Add logic to check for conflicts when adding tasks

junhonglow committed:
Set up Continuous Integration

junhonglow committed:
Fix code styling issues

junhonglow committed:
Improve code quality for test cases

junhonglow committed:
Update JavaDoc comments

junhonglow committed:
Update greet and error messages

junhonglow committed:
Add files via upload

junhonglow committed:
Merge branch 'master' of https://github.com/junhonglow/ip

junhonglow committed:
Update README.md

junhonglow committed:
Update README.md

junhonglow committed:
Update workflow

junhonglow committed:
Update README.md

junhonglow committed:
Update README.md

junhonglow committed:
Add more error handling

junhonglow committed:
Add more JUnit tests

junhonglow committed:
Merge branch 'master' of https://github.com/junhonglow/ip

junhonglow committed:
Update README.md
